File: java/src/test/java/problems1_100/RemoveNthNodeFromEndOfListTest.java
package problems1_100;

import concepts.SinglyListNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class RemoveNthNodeFromEndOfListTest extends SampledTest {
    @Test
    public void test() {
        for (RemoveNthNodeFromEndOfList solution : RemoveNthNodeFromEndOfList.values()) {
            for (RemoveNthNodeFromEndOfListSample sample
                : getSamples(RemoveNthNodeFromEndOfListSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.removeNthFromEnd(
                            SinglyListNode.of(sample.input.head),
                            sample.input.n
                        )
                    ).isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/RemoveNthNodeFromEndOfListTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class RemoveNthNodeFromEndOfListTest : SampledTest() {
    @Test
    fun test() =
        RemoveNthNodeFromEndOfList.entries.forEach { solution ->
            getSamples<Array<RemoveNthNodeFromEndOfListSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.removeNthFromEnd(SinglyListNode.of(*input.head), input.n))
                        .isEqualTo(SinglyListNode.of(*output))
                }
        }
}
